dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -588,4 +620,71 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
                 "Removing bundles in a separated process, the result of the operation will be notified")).build();
     } // deleteAllSuccess.
 
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, MediaType.TEXT_PLAIN})
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public final Response uploadBundle(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+            FormDataMultiPart multipart,
+            @Context UriInfo uriInfo, @QueryParam("sync") boolean sync) throws DotPublisherException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        for (final BodyPart part : multipart.getBodyParts()) {
+
+            InputStream inputStream = (part.getEntity() instanceof InputStream) ? InputStream.class
```

Comment:
Codacy found an issue: [Ensure that resources like this boolean object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)